codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/nameresolver/Dafny.java

```diff
@@ -92,6 +97,74 @@ public static String datatypeConstructorCreate(String name, boolean isRecordType
         return "create_" + DafnyNameResolverHelpers.dafnyCompilesExtra_(name);
     }
 
+    private static final DafnyVersion NEEDS_TYPE_DESCRIPTORS_WHEN_CONSTRUCTING_DATATYPES = new DafnyVersion(4, 2, 0);
```

Comment:
Nit: Shouldn't this be a collection? 
Such that multiple Dafny versions can fit this bill?

Update: OH, Dafnys greater than or equal to this will need it.

```suggestion
    //  Dafnys greater than or equal to this will need Type Descriptors for constructing datatypes  
    private static final DafnyVersion NEEDS_TYPE_DESCRIPTORS_WHEN_CONSTRUCTING_DATATYPES = new DafnyVersion(4, 2, 0);
```